File: examples/js/stream.ts
import "dotenv/config";

import { KovaTTSClient } from "@kova-ai/tts";

const client = new KovaTTSClient({
  apiKey: process.env.KOVA_API_KEY!,
});

for await (const event of client.streamTTS({
  text: "Hello world.",
  voice: process.env.KOVA_TEST_VOICE ?? "cal",
  timestamps: true,
  normalize_text: true,
  response_format: {
    encoding: "pcm"
  }
})) {
  if (event.type === "audio") {
    console.log(`received ${event.audio.byteLength} audio bytes`);
  } else {
    console.log(event.words.join(" "));
  }
}

File: examples/python/stream.py
import asyncio
import os

from env import load_dotenv
from kova_tts import KovaTTSClient

load_dotenv()


async def main() -> None:
    client = KovaTTSClient(
        api_key=os.environ["KOVA_API_KEY"],
    )

    async for event in client.stream_tts(
        text="Hello world.",
        voice=os.environ.get("KOVA_TEST_VOICE", "cal"),
        timestamps=True,
        normalize_text=True,
    ):
        if event.type == "audio":
            print(f"received {len(event.audio)} audio bytes")
        else:
            print(" ".join(event.words))


asyncio.run(main())
